code/modules/mob/living/living.dm

```diff
@@ -426,6 +426,27 @@
 
 		set_pull_offsets(M, state)
 
+/// Adds pass flags and signal to smash through windows as you fly through the air
+/mob/living/proc/start_window_flight(duration = 1.5 SECONDS)
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, PROC_REF(flying_window_smash))
```

Comment:
I think this would work better as a element/component than a signal registered on itself.

Reply:
sure